src/analysis/retail/paladin/shared/HolyPowerTracker.ts

```diff
@@ -1,18 +1,20 @@
 import SPELLS from 'common/SPELLS';
 import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
 import ResourceTracker from 'parser/shared/modules/resources/resourcetracker/ResourceTracker';
+import { Options } from 'parser/core/Analyzer';
+import { CastEvent } from 'parser/core/Events';
 
 class HolyPowerTracker extends ResourceTracker {
-  constructor(...args) {
-    super(...args);
+  constructor(options: Options) {
+    super(options);
     this.resource = RESOURCE_TYPES.HOLY_POWER;
   }
 
-  getAdjustedCost(event) {
-    if (!this.getResource(event).cost) {
+  getAdjustedCost(event: CastEvent) {
+    let cost = this.getResource(event)?.cost ?? 0;
```

Comment:
If you do `?? 0` you don't need the !cost since it could never be nullish

Reply:
The `!cost` is still there so we don't do the cost subtraction in the FIRES_OF_JUSTICE_BUFF check. I didn't want to go too deep with the changes here as I'm not maintaining it.

Reply:
then the `?? 0` is unnecessary, no?

Reply:
Yep!